src/Focusable.ts

```diff
@@ -182,6 +182,10 @@ export class FocusableAPI implements Types.FocusableAPI {
         return this._findElements(true, options) || [];
     }
 
+    findIterator(options: Types.FindIteratorProps): void {
```

Comment:
How does this make things simpler?

```ts
let node;
while(condition(node)) {
  node = findNextFocusable(node);
}
```

Why is the above worse? This iterator pattern does not behave like any iterator pattern that I've ever seen 😨 and would be quite confusing since it relies on an internal API that isn't public either

if we really want an iterator then we should explicitly create a `FocusableIterator` that encapsulates what you want to do here with `.nextNode` `.prevNode`. which honestly feels like it could replace `findNext` and ` findPrev`

Reply:
Because `findNext` and `findPrev` start the process from scratch — they initalize the state, get context and do work to prepare for the actual search. Iterator allows to do that work once and reuse all preparation steps.

Reply:
Perhaps Iterator is the wrong name for this one. This one behave's like JavaScript array's `some()`. I can rename it to `findSome()`.

Reply:
Why does this iterator not return anything? is the only way to get elements from the `find` to use an `onElement` callback ?

Reply:
So, it's not really an iterator. It's a thing that calls callback for every element that it finds and allows to stop search by returning false. I can rename it to findSome and return the resulted array to be consistent with JavaScript's some. Would that be satisfactory?

Reply:
Basically, it's a version of `findAll` that has a callback, can start from custom element and search in any direction and can stop search when needed. I've initially done it in `findAll`, but it seemed too much for a little method finding all, so I have a new one.

Reply:
Renamed and returned the value. Does it look better now?

Reply:
Actually, it's not even `.some()` and more like `.filter()`. But in that context and with the capabilities, I suppose `.findSome()` is a reasonable name.